### PR TITLE
propagate override 

### PR DESCRIPTION
**Pull request by pbischof42**
## Description

Describe what this pull request changes, and why. Include implications for people using this change.
Design decisions and their rationales should be documented in the repo (docstring / ADR), per
[OEP-19](https://open-edx-proposals.readthedocs.io/en/latest/oep-0019-bp-developer-documentation.html), and can be
linked here.

Useful information to include:
- Which edX user roles will this change impact? Common user roles are "Learner", "Course Author",
"Developer", and "Operator".
- Include screenshots for changes to the UI (ideally, both "before" and "after" screenshots, if applicable).
- Provide links to the description of corresponding configuration changes. Remember to correctly annotate these
changes.

## Supporting information

Link to other information about the change, such as Jira issues, GitHub issues, or Discourse discussions.
Be sure to check they are publicly readable, or if not, repeat the information here.

## Testing instructions

Please provide detailed step-by-step instructions for testing this change.

## Deadline

"None" if there's no rush, or provide a specific date or event (and reason) if there is one.

## Other information

Include anything else that will help reviewers and consumers understand the change.
- Does this change depend on other changes elsewhere?
- Any special concerns or limitations? For example: deprecations, migrations, security, or accessibility.
- If your [database migration](https://openedx.atlassian.net/wiki/spaces/AC/pages/23003228/Everything+About+Database+Migrations) can't be rolled back easily.

### CODE REVIEW

File: cms/djangoapps/contentstore/views/item.py [hunk 1]
```diff
@@ -692,6 +695,9 @@ def _save_xblock(user, xblock, data=None, children_strings=None, metadata=None,
             modulestore().publish(xblock.location, user.id)
 
         # Note that children aren't being returned until we have a use case.
+
+        _propagate_library_updates(xblock.location.course_key,user.id)
```
Comment (jcohen28) [inline cms/djangoapps/contentstore/views/item.py]: nit: spacing
```suggestion
        _propagate_library_updates(xblock.location.course_key, user.id)
```

File: cms/djangoapps/contentstore/views/item.py [hunk 2]
```diff
@@ -89,6 +89,9 @@
 NEVER = lambda x: False
 ALWAYS = lambda x: True
 
+@pluggable_override("PROPAGATE_LIBRARY_UPDATES")
+def _propagate_library_updates(library_key_string, user_id):
```
Comment (ianj-lewis) [inline cms/djangoapps/contentstore/views/item.py]: An alternative to creating a separate function here is to override `_save_xblock`, get its return value in the override function via the `prev_fn` arg, and perform the required logic there prior to returning:

```
def override_save_xblock(_save_xblock, *args, **kwargs):
    response = _save_xblock(*args, **kwargs)
    # Your logic here
    return response
```

This way we only need to include one line of changes in the edx repo, `@pluggable_override("OVERRIDE_SAVE_XBLOCK")`. Although on the other hand, defining a custom function tamps down on the scope of what we're overriding / allows us to handle only the arguments we need. @tramck do you have an opinion?

https://github.com/openedx/edx-django-utils/blob/2c975229cbbbfc07f3320b72cbb91f208957f2f5/edx_django_utils/plugins/pluggable_override.py#L26-L28

File: cms/djangoapps/contentstore/views/item.py [hunk 3]
```diff
@@ -89,6 +89,9 @@
 NEVER = lambda x: False
 ALWAYS = lambda x: True
 
+@pluggable_override("PROPAGATE_LIBRARY_UPDATES")
```
Comment (ianj-lewis) [inline cms/djangoapps/contentstore/views/item.py]: Pluggable override names should always be prefixed with `OVERRIDE_`

```suggestion
@pluggable_override("OVERRIDE_PROPAGATE_LIBRARY_UPDATES")
```

**Comment by ianj-lewis** [comment]
Anytime we're making changes to the edx repo, we should include a `@medality_custom` comment so we can easily find where we've made our own modifications, here's an example: https://github.com/Medality-Health/edx-platform/blob/4ac6887359d23a93eefe9f20aff4639dd0dda164/lms/djangoapps/course_blocks/transformers/library_content.py#L27-L43

File: cms/djangoapps/contentstore/views/item.py [hunk 4]
```diff
@@ -527,8 +528,8 @@ def _update_with_callback(xblock, user, old_metadata=None, old_content=None):
     # Update after the callback so any changes made in the callback will get persisted.
     return modulestore().update_item(xblock, user.id)
 
-
-def _save_xblock(user, xblock, data=None, children_strings=None, metadata=None, nullout=None,  # lint-amnesty, pylint: disable=too-many-statements
+@pluggable_override("OVERRIDE_SAVE_XBLOCK")
+def _save_xblock(user, xblock, method, data=None, children_strings=None, metadata=None, nullout=None,  # lint-amnesty, pylint: disable=too-many-statements
```
Comment (tramck) [inline cms/djangoapps/contentstore/views/item.py]:
```suggestion
def _save_xblock(user, xblock, data=None, children_strings=None, metadata=None, nullout=None,  # lint-amnesty, pylint: disable=too-many-statements
```

**Review by tramck — APPROVED** [review]

**Pull request merged**